druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java

```diff
@@ -2123,6 +2123,51 @@ public Void apply(ResultSet resultSet) {
     sql(sql).returnsUnordered("C=60").queryContains(druidChecker("'queryType':'timeseries'"));
   }
 
+  /** Tests that the aggregate in the druid query
+   * is of type hyperUnique rather than cardinality */
+  @Test public void testHyperUniqueAggregateProduced() {
+    String sql = "select count(distinct \"user_unique\") as users from \"wiki\"";
+    String aggString = "{'type':'hyperUnique','name':'USERS','fieldName':'user_unique'}";
+    sql(sql, WIKI)
+      .queryContains(druidChecker(aggString));
+  }
+
+  /** Tests that non distinct count aggregates don't use the hyperUnique type
+   * for metrics with type hyperUnique */
+  @Test public void testNonDistinctCountAggregateProducedForHU() {
+    String sql = "select count(\"user_unique\") as users from \"wiki\"";
+    String aggString = "{'type':'count','name':'USERS','fieldName':'user_unique'}";
```

Comment:
this is not correct, there is no such aggregator in druid. As per the [doc](druid.io/docs/latest/querying/aggregations.html#count-aggregator), the count aggregator dose count number of lines. In order to do a count(dim) you need to use nested group by (first group by the dimension then do the count).